docs/project/versioning.md

```diff
@@ -0,0 +1,269 @@
+# Toolchain and language versioning
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/4105)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Overview](#overview)
+-   [Major version increments](#major-version-increments)
+    -   [Breaking changes](#breaking-changes)
+    -   [Exclusions to what constitutes a breaking change](#exclusions-to-what-constitutes-a-breaking-change)
+-   [Minor version increments](#minor-version-increments)
+-   [Patch version increments](#patch-version-increments)
+    -   [Examples:](#examples)
+-   [Pre-release versions](#pre-release-versions)
+    -   [Release qualification pre-releases: `MAJOR.MINOR.PATCH-rc.N`](#release-qualification-pre-releases-majorminorpatch-rcn)
+    -   [Incremental development versions: `MAJOR.MINOR.PATCH-0.{nightly,dev}.N`](#incremental-development-versions-majorminorpatch-0nightlydevn)
+        -   [Nightly pre-release versions](#nightly-pre-release-versions)
+        -   [Development pre-release versions](#development-pre-release-versions)
+-   [Relevant proposal](#relevant-proposal)
+
+<!-- tocstop -->
+
+## Overview
+
+Carbon uses a single versioning scheme across both the language itself and
+toolchain, including the standard library, compiler, linker, and all development
+tools released by the main Carbon project.
+
+The scheme conforms to and is closely based on Semantic Versioning (SemVer):
+
+-   Carbon versions: `MAJOR.MINOR.PATCH`
+-   `MAJOR` increments on backwards incompatible changes, including a
+    deprecation that might trigger a build-breaking warning.
+-   `MINOR` increments only expected during early development with a major
+    version of `0`.
+-   `PATCH` increments represent bug fixes only.
```

Comment:
For major and patch, we say what the increments mean, whereas for minor we don't say what they mean, only that we won't do them much. If we're going to explain how to understand the version numbers rather than deferring to the SemVer spec, it would be good to say what the minor portion of the version means too.

Something like this would capture what I understood from our prior discussion:
```suggestion
-   `MAJOR` increments on backwards incompatible changes, including a
    deprecation that might trigger a build-breaking warning.
-   `MINOR` increments on backwards incompatible changes prior to
    reaching a major version of `1`, and is expected to remain at `0`
    afterwards.
-   `PATCH` increments represent bug fixes only.
```

It might be clearer to swizzle this around:
```suggestion
-   Releases with backwards incompatible changes, including a
    deprecation that might trigger a build-breaking warning,
    increment `MAJOR` after reaching the `1.0` milestone, and
    increment `MINOR` before then.
-   Releases with only backwards compatible changes are not
    expected to happen.
-   Releases containing only bug fixes increment `PATCH`.
```